### PR TITLE
[demultiplexer] adapt unit tests to not use the global aggregator instance

### PR DESCRIPTION
**Pull request by remeh**
### What does this PR do?

In order to fix race conditions in some `pkg/aggregator` unit tests, this pull-request is adapting several unit tests to use a local instance of the `Demultiplexer` (and thus of the `BufferedAggregator`).

Since it is planned in the future to get rid of the global instance of the aggregator and of the demultiplexer, this would have to be done at some point anyway.

### Additional Notes

The race never occurred on our CI nor CircleCI, it seems to only happen while running in AppVeyor.

### Possible Drawbacks / Trade-offs

None.

### Describe how to test/QA your changes

Unit tests are all green.

### Reviewer's Checklist

- [x] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [x] The appropriate `team/..` label has been applied, if known.
- [x] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [x] Changed code has automated tests for its functionality.
- [x] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [x] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [x] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [x] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/aggregator/sender_test.go [hunk 1]
```diff
@@ -40,46 +40,51 @@ func initSender(id check.ID, defaultHostname string) (s senderWithChans) {
...
+func testDemux() *AgentDemultiplexer {
+	opts := DefaultDemultiplexerOptions(nil)
+	opts.DontStartForwarders = true
+	demux := initAgentDemultiplexer(opts, defaultHostname)
+	return demux
+}
+
 func TestGetDefaultSenderReturnsSameSender(t *testing.T) {
-	// this test IS USING globals (demultiplexerInstance holding the senders)
+	// this test not using anything global
```
Comment (ogaca-dd) [inline pkg/aggregator/sender_test.go:51]: Is this message still relevant as most tests do not use anything global?

File: pkg/aggregator/sender_test.go [hunk 2]
```diff
@@ -145,25 +145,25 @@ func TestGetAndSetSender(t *testing.T) {
...
 
-	sender, err := GetSender(checkID1)
+	sender, err := demux.GetSender(checkID1)
 	assert.Nil(t, err)
 	assert.Equal(t, testCheckSender, sender)
 }
 
 func TestGetSenderDefaultHostname(t *testing.T) {
-	// this test IS USING globals (demultiplexerInstance holding the senders)
+	// this test not using anything global
```
Comment (ogaca-dd) [inline pkg/aggregator/sender_test.go:157]: What about `aggregatorInstance`?

**Review by ogaca-dd — APPROVED** [review]
LGTM

File: pkg/aggregator/sender_test.go (hunk 2 above)
Comment (remeh) [reply pkg/aggregator/sender_test.go:157]: It is read from the local demultiplexer L161:

```
	aggregatorInstance := demux.Aggregator()
```

File: pkg/aggregator/sender_test.go (hunk 1 above)
Comment (remeh) [reply pkg/aggregator/sender_test.go:51]: I think I've left a few comments still using globals in other files.
Seeing the history of people fighting with globals in tests, I prefer to keep them for a while if you don't mind.

**Pull request merged**